src/lib/people/team/team.ts

```diff
@@ -0,0 +1,64 @@
+import { assert } from '$lib/assert';
+
+export const committees = [
+    'Executive',
+    'Innovation',
+    'Service',
+    'External Relations',
+    'Membership & Internals',
+    'Branding & Creatives',
+    'Engineering',
+];
+
+export interface Member {
+    name: string;
+    title: string;
+    committee: string;
+    img: string;
+    socials: Record<string, string>;
+}
+
+export default async function getTeam(): Promise<Member[]> {
+    /**
+     * The usage of dynamic imports in this function will have no effect on page load times.
+     * This is because dynamically imported assets will still be bundled on build.
+     * Thus, this will make things more organized over storing the images in "static/".
+     */
+    const teamPromise = import.meta.glob<Member>('./members/*.json', {
+        import: 'default',
+    });
+
+    const team: Member[] = [];
+    for (const path in teamPromise) {
```

Comment:
This can be refactored into a much cleaner version with [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) and [`Object.entries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries).

> [!IMPORTANT]
> In my implementation, I removed the `img` field in the `Member` JSON because we can actually derive the intended file name on the import path anyway.

```js
import { basename } from 'node:path';

const imports = import.meta.glob('$lib/members/*.json');
const promises = Object.entries(imports).map(async ([path, asset]) => {
    // Derive the image name from the original path
    const base = basename(path, '.json');
    const img = await import(`$lib/people/team/images/${base}.webp`).default;

    // See my other suggestion on Zod and Valibot for more details.
    const member = parse(Member, await asset);

    // Type-safety enforced at build-time and compile-time!
    return { ...member, img };
});

// Await all imports in parallel!
const members = await Promise.all(promises);
```

Reply:
I didn't initially consider deriving the intended file name for the import path, but I really like your approach so I'll try adding it to the PR. Thanks for that and the suggested refactor!

That said, I've been having difficulty getting `node:path`, which was used in the provided code, to work as expected. Looking it up, [it doesn't seem to mix particularly well with Vite](https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility). That aside, when I tried using polyfills for it (i.e. [vite-plugin-node-polyfills](https://www.npmjs.com/package/vite-plugin-node-polyfills)), I ran into a problem where it breaks the dev server (`500 Internal Error`) upon _reloading_ the People page. While this doesn't seem to be an issue on build, it seems like it would be annoying to deal with. 

Anyhow, I'll be opting for a normal typescript or regex approach for the filename rather than using `basename` until this issue gets figured out if that's alright.

Reply:
> That said, I've been having difficulty getting node:path, which was used in the provided code, to work as expected.

Ah, I got it. We need to rename the `+page.ts` to `+page.server.ts` so that it only runs in the dev server (i.e., a Node.js environment) and the build step (i.e., a Node.js environment). The `+page.ts` tells SvelteKit to emit client-side bundles, which would indeed break when we import `node:path`. But if we use `+page.server.ts` instead, the `adapter-static` will emit `.json` files in the `build` folder.

## Resources
* [The Missing Guide to Understanding `adapter-static` in SvelteKit](https://khromov.se/the-missing-guide-to-understanding-adapter-static-in-sveltekit/)